services/src/main/java/com/broadleafcommerce/subscriptionoperation/service/DefaultSubscriptionPricingService.java

```diff
@@ -0,0 +1,875 @@
+
+            BigDecimal itemUnitPrice = Optional.ofNullable(subscriptionItem.getItemUnitPrice())
+                    .orElse(BigDecimal.ZERO);
+            BigDecimal itemTotal =
+                    itemUnitPrice.multiply(BigDecimal.valueOf(subscriptionItem.getQuantity()));
+
+            pricingContext.getExistingSubscriptionItemPrices()
+                    .put(itemId, MonetaryUtils.toAmount(itemTotal, pricingContext.getCurrency()));
+        });
+    }
+
+    protected void populateFromCartItem(@lombok.NonNull SubscriptionPricingContext pricingContext,
+            @lombok.NonNull CartItem subscriptionRootItem,
+            @Nullable ContextInfo contextInfo) {
+        pricingContext.setPaymentStrategy(
+                getSubscriptionPaymentStrategy(subscriptionRootItem, contextInfo));
+
+        RecurringPriceDetail recurringPriceDetail = subscriptionRootItem.getRecurringPrice();
+        pricingContext.setPeriodType(recurringPriceDetail.getPeriodType());
+        pricingContext.setPeriodFrequency(recurringPriceDetail.getPeriodFrequency());
+
+        Instant endOfTermsDate = determineEndOfTermsDate(subscriptionRootItem.getTermDurationType(),
+                subscriptionRootItem.getTermDurationLength(), pricingContext);
+        pricingContext.setEndOfTermsDate(endOfTermsDate);
+
+        Instant atypicalNextBillDate =
+                determineAtypicalNextBillDate(subscriptionRootItem, pricingContext, contextInfo);
+        pricingContext.setAtypicalNextBillDate(atypicalNextBillDate);
+    }
+
+    protected String getSubscriptionPaymentStrategy(@lombok.NonNull CartItem subscriptionRootItem,
+            @Nullable ContextInfo contextInfo) {
+        // TODO: Refine this implementation & "subscriptionPaymentStrategy" name once catalog
+        // representation evolves
+        if (MonetaryUtils.isGreaterThanZero(subscriptionRootItem.getSubtotal())) {
+            return DefaultSubscriptionPaymentStrategy.IN_ADVANCE.name();
+        } else {
+            return DefaultSubscriptionPaymentStrategy.POSTPAID.name();
+        }
+    }
+
+    protected Instant determineEndOfTermsDate(@lombok.NonNull String termDurationType,
+            @lombok.NonNull Integer termDurationLength,
+            @lombok.NonNull SubscriptionPricingContext pricingContext) {
+        Instant today = pricingContext.getSubscriptionActionDate()
+                .truncatedTo(ChronoUnit.DAYS);
+        LocalDateTime ldt = today.atZone(ZoneOffset.UTC).toLocalDateTime();
+
+        if (DefaultTermDurationType.isDays(termDurationType)) {
+            ldt.plusDays(termDurationLength);
+        } else if (DefaultTermDurationType.isWeeks(termDurationType)) {
+            ldt.plusWeeks(termDurationLength);
+        } else if (DefaultTermDurationType.isMonths(termDurationType)) {
+            ldt.plusMonths(termDurationLength);
+        } else if (DefaultTermDurationType.isYears(termDurationType)) {
+            ldt.plusYears(termDurationLength);
+        }
```

Comment:
Javadocs say
> Returns a copy of this {@\code LocalDateTime} with the specified number of days added.
> ...
> This instance is immutable and unaffected by this method call.

So pretty sure you need to reassign `ltd` to the result

```suggestion
        if (DefaultTermDurationType.isDays(termDurationType)) {
             ltd = ldt.plusDays(termDurationLength);
        } else if (DefaultTermDurationType.isWeeks(termDurationType)) {
             ltd = ldt.plusWeeks(termDurationLength);
        } else if (DefaultTermDurationType.isMonths(termDurationType)) {
            ldt.plusMonths(termDurationLength);
        } else if (DefaultTermDurationType.isYears(termDurationType)) {
            ltd = ldt.plusYears(termDurationLength);
        }
```